### PR TITLE
gatfruit shoots peas

### PR DESCRIPTION
**Pull request by mc-oofert**
## About The Pull Request
basically #64128 slightly redone (with permission)

gatfruit now shoots peas; 
(injects assuming hit zone is unblocked and this assumes you havent modified the plants reagent genes)
at 0 potency to god knows what potency a single pea (bullet) deals 5 brute and injects 1u of phytotoxins
at 50 potency it fires 10 brute and injects 15u of phytotoxin
at 100 potency it fires 15 brute and injects 30u of phytotoxin
phytotoxin is the Toxin chemical but with no liver damage, it is not particularly strong but serves to recreate the original PRs 5 tox with possibility of actually modifying that

2024-11-20 17_04_00-Window

upon peeling the fruit its reagents are spread among the revolvers 7 casings

also i attempted to make the revolver look more plant-like but im pretty sure i screwed up
image

## Why It's Good For The Game

a bullet of 357 deals 60 brute (not counting falloff (which i dont think it gets) and armor)
there are 7 bullets of 357 in a revolver so 420 brute in 1 revolver, and it basically takes 200 brute to kill someone outright if oxygen and other damage doesnt do it first
at 10 yield (or whatever max yield is i cant really remember) thats 10 revolvers, so 10 x 420 so 4200 in a bag brute per harvest (again 200 brute if you decide to shoot until something dies so you can kill like 21 people if you dont miss in a single yield 10 harvest)
thats pretty busted for something you can achieve fast by making REALLY easy chems after getting the seeds from exodrones or lavaland

with this change its still a viable weapon but not a supremely lethal traitor weapon, and also enables some botany gaming if you make it contain felinid mutation toxin or something

## Changelog
:cl:
balance: A mutation in gatfruit seeds has led to a drastic alteration in the observable traits of the plant, which now fires hardened peas that deal less damage, but poison the target. Additionally, its poison can be, with some botanical engineering, replaced with whatever you wish.
/:cl:

### CODE REVIEW

**Comment by SmArtKar** [comment]
Assuming with this you'll revert #87980's gatfruit removal, right? The map itself is pretty nice, if its not a GBJ with no gatfruits i'll probably get merged

**Comment by mc-oofert** [comment]
> Assuming with this you'll revert 87980's gatfruit removal, right? The map itself is pretty nice, if its not a GBJ with no gatfruits i'll probably get merged

i will revert that PRs spawner change if this PR gets merged before that one

File: code/modules/projectiles/projectile/bullets/revolver.dm [hunk 1]
```diff
@@ -152,3 +152,38 @@
...
+				target.visible_message(span_danger("\The [src] is deflected!"), \
+									   span_userdanger("You are protected against \the [src]!"))
+
+	. = ..(target, blocked)
+	if(reagents & NO_REACT) //first imapct
+		reagents.flags &= ~(NO_REACT)
+		reagents.handle_reactions()
+
+/obj/projectile/bullet/pea/on_hit(mob/living/target, blocked = 0, pierce_hit)
+	if(iszombie(target)) // https://www.youtube.com/watch?v=ssZoq1eUK-s
```
Comment (Sadboysuss) [inline code/modules/projectiles/projectile/bullets/revolver.dm:187]: this is great

File: sound/items/weapons/peashoot.ogg
Comment (Sadboysuss) [inline sound/items/weapons/peashoot.ogg:1]: I like the sound, but where is the attribution?

**Comment by Loafin34** [comment]
crossbreed with chilly peppers for ice pea?

File: sound/items/weapons/peashoot.ogg
Comment (mc-oofert) [reply sound/items/weapons/peashoot.ogg:1]: idk where the sound is from came with the other pr

**Comment by mc-oofert** [comment]
> crossbreed with chilly peppers for ice pea?

just crosspollinate

File: code/modules/projectiles/ammunition/ballistic/revolver.dm [hunk 2]
```diff
@@ -67,3 +67,35 @@
...
+		return
+	loaded_projectile.damage = damage
+
+/obj/item/ammo_casing/pea/attack_self(mob/user)
+	. = ..()
+	if(isnull(loaded_projectile))
+		return
+	var/obj/item/food/grown/peas/peas = new(user.drop_location())
+	user.put_in_hands(peas)
+	to_chat(user, span_notice("You seperate the pea from the case."))
```
Comment (MrMelbert) [inline code/modules/projectiles/ammunition/ballistic/revolver.dm:99]:
```suggestion
	to_chat(user, span_notice("You separate [peas] from [src]."))
```

File: code/modules/projectiles/projectile/bullets/revolver.dm [hunk 3]
```diff
@@ -152,3 +152,38 @@
...
+			if(target.can_inject(target_zone = def_zone)) // Pass the hit zone to see if it can inject by whether it hit the head or the body.
+				..()
+				reagents.trans_to(target, reagents.total_volume, methods = INJECT)
+				return BULLET_ACT_HIT
+			else
+				blocked = 100
+				target.visible_message(span_danger("\The [src] is deflected!"), \
+									   span_userdanger("You are protected against \the [src]!"))
+
+	. = ..(target, blocked)
```
Comment (MrMelbert) [inline code/modules/projectiles/projectile/bullets/revolver.dm:181]:
```suggestion
	. = ..()
```

File: code/modules/projectiles/projectile/bullets/revolver.dm [hunk 4]
```diff
@@ -152,3 +152,38 @@
...
+	if(reagents & NO_REACT) //first imapct
+		reagents.flags &= ~(NO_REACT)
+		reagents.handle_reactions()
+
+/obj/projectile/bullet/pea/on_hit(mob/living/target, blocked = 0, pierce_hit)
```
Comment (MrMelbert) [inline code/modules/projectiles/projectile/bullets/revolver.dm:186]: You have two `/obj/projectile/bullet/pea/on_hit`s defined

File: code/modules/projectiles/projectile/bullets/revolver.dm [hunk 5]
```diff
@@ -152,3 +152,38 @@
...
+	if(istype(target))
+		if(blocked != 100) // not completely blocked
+			if(target.can_inject(target_zone = def_zone)) // Pass the hit zone to see if it can inject by whether it hit the head or the body.
+				..()
+				reagents.trans_to(target, reagents.total_volume, methods = INJECT)
+				return BULLET_ACT_HIT
+			else
+				blocked = 100
+				target.visible_message(span_danger("\The [src] is deflected!"), \
+									   span_userdanger("You are protected against \the [src]!"))
```
Comment (MrMelbert) [inline code/modules/projectiles/projectile/bullets/revolver.dm:179]: This whole part of the proc could be cleaned up a bit - early returns etc

File: code/modules/hydroponics/grown/gatfruit.dm [hunk 6]
```diff
@@ -17,15 +17,26 @@
...
 	wine_power = 90 //It burns going down, too.
+
+/obj/item/food/grown/shell/gatfruit/generate_trash(atom/location)
+	//if you set this to anything but the revolver i will find you... and... downvote your pr...
+	var/obj/item/gun/ballistic/revolver/peashooter/gun = new trash_type(location || drop_location())
+	var/potency_percentage = CLAMP01(seed.potency / 100)
+	for(var/obj/item/ammo_casing/pea/casing as anything in gun.magazine.ammo_list())
+		casing.damage = floor(max(5, LERP(5, casing.max_damage, potency_percentage)))
+		if(reagents.total_volume)
+			reagents.trans_to(casing, reagents.total_volume / gun.magazine.max_ammo)
```
[the remark this answers is not in the document]
Comment (mc-oofert) [reply code/modules/hydroponics/grown/gatfruit.dm:40]: all four chems can be used for reactions to create whatever; removing these from the casings will prevent any reaction that uses these

if a dude wants a bullet filled with exclusively toxin then they need the geneshears

[the remark this answers is not in the document]
Comment (MrMelbert) [reply code/modules/hydroponics/grown/gatfruit.dm:40]: They would react instantly on harvesting the plant anyways

Unless you mean someone having one gun filled with x reagent and a second gun fixed with y reagent, which in that case now I'm scared that botanists have access to cryo darts

[the remark this answers is not in the document]
Comment (mc-oofert) [reply code/modules/hydroponics/grown/gatfruit.dm:40]: if we were to remove carbon, sulfur, nitrogen, potassium at runtime from the harvested fruit then they cant make a revolver that shoots salicylic acid for example (uses carbon in its recipe) or whatever else

**Comment by carlarctg** [comment]
i was coming in on this pr ready to hate on it but this kinda slaps actually

only problem is the pea shooter not looking like a pea shooter (also the ammo not looking like peas)

**Comment by mc-oofert** [comment]
> i was coming in on this pr ready to hate on it but this kinda slaps actually
> 
> only problem is the pea shooter not looking like a pea shooter (also the ammo not looking like peas)

im not a spriter

**Comment by SmArtKar** [comment]
> > i was coming in on this pr ready to hate on it but this kinda slaps actually
> > only problem is the pea shooter not looking like a pea shooter (also the ammo not looking like peas)
> 
> im not a spriter

not a spritainer but it looks pretty good to me, soulful (in a good way, not a sovl way)

**Comment by EricZilla** [comment]
The peavolver is real, love this

**Comment by DrAmazing343** [comment]
This is huge gatbros

**Comment by necromanceranne** [comment]
I kinda love the fact it benefits from adding chems to the plant. Its actually integrated into botany properly.

**Comment by the-orange-cow** [comment]
Out of curiosity, does the injection effect pierce thick clothing/modsuits?

**Comment by sylvia-from-fulp-station** [comment]
this is not a viable weapon it sucks without chemistry powergaming

**Comment by Loafin34** [comment]
its a disposable weapon tf did you want

**Comment by sylvia-from-fulp-station** [comment]
i want the 357 LOL

**Comment by necromanceranne** [comment]
> this is not a viable weapon it sucks without chemistry powergaming

It's _botany_, that's literally what botany does.

**Comment by Holinka4ever** [comment]
I want to make little critical reception about these pr, to point crucial parts of it, and then make conclusion with recommendations.
_All opinion here is subjective_

**Pros**
**1.** No more an antag weapon  in public use, that can deal 60 damage per shot, powergamers, validers and other antags but tot botanist, are in shambles. 
**2.** New shenigans with botanics, make it actually valid for botanist and their powertripping experience to scale peashooter into literally heaven.

**Cons**
**1.** Its turning thing literally useless for any jobs and antags, if you are not tot botanist. Lets be clear, you can get seeds from: lava, xenobio, exploration drones. Roles that are not botanists, doesnt have resources as botanists, and at all, it will just feel weird
**2.** Accessability, as i said, you cant get seeds if you are not botanist, and it becomes more nishe for tot botanist , than sci tot item that working with gold extract. Cause even they both need experience to work, atleast as sci you can hide your spider army in near maints, hiding one hydroponic in botanist room will raise suspicious.
**3.** Non effective for actual tot botanist, as we know, tot botanist just feel station with kudzu,  killer tomatoes and smallcrossbow, and them are much safer to pull, rather than ice peashooter.
**4.** Populazation of laser weapons instead of balistics, thats actually, disgusting how coders trying to put into basis that "only one weapon that can resolve problem" and thats third degree burn wounds. Very not cool.

In conclusion i would say that nerf is justified, actually not, i see that people gonna blame rework because it became too nishe to work with. Like, not even tot botanist will actually use gatfruit due of a lot of time to maintain it to instanly being round removed due of being valid.
For recomendations, to save it from absense, turn it into pea mutation instead, rather than tot role seed package, like, and give it same traits as nettles, aka you cant use it without botanist gloves.
If nerf turn them weak, why its only rng obtainable and difficult to maintain with?

P.S.
Due of people looking how coders nerf stuff, they understand that, instead of crucial looking of which part is broken by, they just doing big axe swing to just kill it, rather to fix. That was with, that was with jetpacks, that is also with that pr. It just kills, nothing more, and nithing less.

Peace out.

**Comment by Holinka4ever** [comment]
Also we should include that getting gatfruit is literally rng, and nerfing staff that need to work out is disgusting way to fix it.
Exploration drones also stop being valid, cause only gatfruit let em be on water slide, now these mechanic will be absent.

**Comment by mc-oofert** [comment]
> Out of curiosity, does the injection effect pierce thick clothing/modsuits?

no

**Comment by mc-oofert** [comment]
> Also we should include that getting gatfruit is literally rng, and nerfing staff that need to work out is disgusting way to fix it. Exploration drones also stop being valid, cause only gatfruit let em be on water slide, now these mechanic will be absent.

exodrones failed literally as soon as they got merged literally nobody touched them 4 rounds after it got merged nor nobody wanted to work with its code

**Comment by Livrah** [comment]
Honestly I'd prefer to just remove gatfruits from the game instead of this. 
We got "TG classic" removal RP #87980 (which I agree with), followed by "whatever "better" solution" gets players upvote. Usually it got coded by other players, not the original PR author.
I hope this joke won't be locked behind contraband or premium seeds.

**Comment by mc-oofert** [comment]
> Honestly I'd prefer to just remove gatfruits from the game instead of this. We got "TG classic" removal RP 87980 (which I agree with), followed by "whatever "better" solution" gets players upvote. Usually it got coded by other players, not the original PR author. I hope this joke won't be locked behind contraband or premium seeds.

everytime you remove something busted some dimwits will spam "improve dont remove" and if you do improve the same people will try to find every single issue this is not a winnable issue

**Comment by mc-oofert** [comment]
> **Cons** **1.** Its turning thing literally useless for any jobs and antags, if you are not tot botanist. Lets be clear, you can get seeds from: lava, xenobio, exploration drones. Roles that are not botanists, doesnt have resources as botanists, and at all, it will just feel weird **2.** Accessability, as i said, you cant get seeds if you are not botanist, and it becomes more nishe for tot botanist , than sci tot item that working with gold extract. Cause even they both need experience to work, atleast as sci you can hide your spider army in near maints, hiding one hydroponic in botanist room will raise suspicious. **3.** Non effective for actual tot botanist, as we know, tot botanist just feel station with kudzu, killer tomatoes and smallcrossbow, and them are much safer to pull, rather than ice peashooter. **4.** Populazation of laser weapons instead of balistics, thats actually, disgusting how coders trying to put into basis that "only one weapon that can resolve problem" and thats third degree burn wounds. Very not cool.
> 
> In conclusion i would say that nerf is justified, actually not, i see that people gonna blame rework because it became too nishe to work with. Like, not even tot botanist will actually use gatfruit due of a lot of time to maintain it to instanly being round removed due of being valid. For recomendations, to save it from absense, turn it into pea mutation instead, rather than tot role seed package, like, and give it same traits as nettles, aka you cant use it without botanist gloves. If nerf turn them weak, why its only rng obtainable and difficult to maintain with?

we are not giving roundstart botany ranged deathnettles

**Comment by Livrah** [comment]
> > Honestly I'd prefer to just remove gatfruits from the game instead of this. We got "TG classic" removal RP 87980 (which I agree with), followed by "whatever "better" solution" gets players upvote. Usually it got coded by other players, not the original PR author. I hope this joke won't be locked behind contraband or premium seeds.
> 
> everytime you remove something busted some dimwits will spam "improve dont remove" or some form of thereof and if you do improve the same people will try to find every single issue so this is not a winnable issue

Screenshot_543
please take bit more seconds to read. I think it should be removed from the most common source (and I bet the only source you met).

**Comment by Holinka4ever** [comment]
> we are not giving _roundstart_ botany ranged nettles
Alright then, put it into emagged seed vendor

> also just cooperate with cargo for gatfruit i mean its like having roleplay in a roleplay game??
Roleplay is not a wildcard for any problem that game might have, like, you cant convice cargo tech to spend whole shift to do exodrones, that will not benefit nor him, nor cargo

**Comment by mc-oofert** [comment]
> > we are not giving _roundstart_ botany ranged nettles
> > Alright then, put it into emagged seed vendor
> 
> > also just cooperate with cargo for gatfruit i mean its like having roleplay in a roleplay game??

Roleplay is not a wildcard for any problem that game might have, like, you cant convice cargo tech to spend whole shift to do exodrones, that will not benefit nor him, nor cargo

pay him 2000cr or a cut of your new potency 100 revolvers

**Comment by Sadboysuss** [comment]
Would be nice if you kept the origibal gatfruit as admin spawn

**Comment by Deultus** [comment]
Can you put this in a hacked vending machine? I think it's appropriate for this topic. let's say initially you just get a bush with cartridges, and mutate a gotfruit from it

**Comment by CRITAWAKETS** [comment]
everyone who thinks this is a nerf should actually read the PR and know how botany works, with just a smidge of applying botany this thing can one-shot anyone who's not coked up to the gills on multiver.

i think you can also syringe reagents into this, which would make it not even require botany for such a thing.

**Comment by Deultus** [comment]
> everyone who thinks this is a nerf should actually read the PR and know how botany works, with just a smidge of applying botany this thing can one-shot anyone who's not coked up to the gills on multiver.
> 
> i think you can also syringe reagents into this, which would make it not even require botany for such a thing.

It won't even be more dangerous than a man with a syringe. With a shot, you enter only about 4 units

**Comment by Holinka4ever** [comment]
> everyone who thinks this is a nerf should actually read the PR and know how botany works, with just a smidge of applying botany this thing can one-shot anyone who's not coked up to the gills on multiver.
> 
> i think you can also syringe reagents into this, which would make it not even require botany for such a thing.

You should know that how botanics ever work, basicly, its much worse than you see.
Cause, we cant assume how these weapon gonna work as plant without actual, but it will be too powerfull if it literally inject you with 30 of anything per shot.
Friend of mine assumn that that reagents inside are for gun, not for bullets.
And it gonna deal in full potency and as own...
20 damage per shot, rather than 45 as people think. Its just killing weapon itself.

We need a video how these weapon damage and actually work, rather than writing and screenshots

**Comment by mc-oofert** [comment]
oh yeah i should make gatfruit fruit not injectable

**Comment by mc-oofert** [comment]
> You should know that how botanics ever work, basicly, its much worse than you see. Cause, we cant assume how these weapon gonna work as plant without actual, but it will be too powerfull if it literally inject you with 30 of anything per shot. Friend of mine assumn that that reagents inside are for gun, not for bullets. And it gonna deal in full potency and as own... 20 damage per shot, rather than 45 as people think. Its just killing weapon itself.

i have literally not ever used the number 45 in the description to refer to damage and if you think i have then you need to reread it

**Comment by mc-oofert** [comment]
> everyone who thinks this is a nerf should actually read the PR and know how botany works, with just a smidge of applying botany this thing can one-shot anyone who's not coked up to the gills on multiver.

wear pierceproof clothing

**Comment by Holinka4ever** [comment]
> > You should know that how botanics ever work, basicly, its much worse than you see. Cause, we cant assume how these weapon gonna work as plant without actual, but it will be too powerfull if it literally inject you with 30 of anything per shot. Friend of mine assumn that that reagents inside are for gun, not for bullets. And it gonna deal in full potency and as own... 20 damage per shot, rather than 45 as people think. Its just killing weapon itself.
> 
> i have literally not ever used the number 45 in the description to refer to damage and if you think i have then you need to reread it

I know you didnt type about 45 damage, but people whould think that "Oh cool, 45 damage where 30 is poison!!!"
Cause you told "in 100 potency it shoots 15 damage and injects 30 units of phytoxin" people would think that toxin deals 1 damage.

**Comment by mc-oofert** [comment]
god knows how toxin metabolism works

**Comment by sylvia-from-fulp-station** [comment]
> > this is not a viable weapon it sucks without chemistry powergaming
> 
> It's _botany_, that's literally what botany does.

have fun playing with your plants bro im gonna wrench the det gun and shoot myself in the head

**Comment by Holinka4ever** [comment]
There is also a third way of resolving these problem, just add these into seed vault, but you still can get original gatfruit with drones, xenobio and tot botanist, everyone happy!
I would rather recommend it, because peashootas are working in harmony with pod persons.

**Comment by mc-oofert** [comment]
> There is also a third way of resolving these problem, just add these into seed vault, but you still can get original gatfruit with drones, xenobio and tot botanist, everyone happy! I would rather recommend it, because peashootas are working in harmony with pod persons.

infinite 357 revolvers suck

**Review by Wallemations — APPROVED** [review]
https://github.com/user-attachments/assets/57120d92-0a13-43a5-9191-62c38a33100e

**Comment by Yttriums** [comment]
Just wanna say I think this is a sick change, long felt like gatfruit was out of place and I think this is a cool evolution of the idea. IDK about balance, but that can all be adjusted over time.

File: code/modules/projectiles/projectile/bullets/revolver.dm (hunk 1 above)
Comment (Yttriums) [reply code/modules/projectiles/projectile/bullets/revolver.dm]: Adding zombie mutation toxin to my gatfruit

**Comment by Blacklist897** [comment]
Im deeply scared of what someone like melbert or ari ixo could do with this

**Comment by MrMelbert** [comment]
This is actually more op than the original gatfruit because you can just have one with water, one with potassium, dual wield -> kill, but I'm keeping that on the down low

**Comment by mc-oofert** [comment]
> This is actually more op than the original gatfruit because you can just have one with water, one with potassium, dual wield -> kill, but I'm keeping that on the down low

you can already do this with literally any other botany plant or shotgun darts or syringes

**Comment by Yttriums** [comment]
Yeah I think this isn't significantly more powerful than nettles with throwing arm or the traitor dartgun, both of which are much easier to get. Still, the opportunity to hand out mindbreaker/mushroom hallucinogen/space drugs guns to assistants is very enticing.

**Comment by Eronymun** [comment]
>attempts to nerf the revolver plant
>gives it the ability to inject through botany
in the hands of an experienced botanist, you just made something far more terrifying than a 357

**Comment by mc-oofert** [comment]
> > attempts to nerf the revolver plant
> > gives it the ability to inject through botany
> 
> in the hands of an experienced botanist, you just made something far more terrifying than a 357

to be fair the people that usually grow this plant are usually cargo staff or assistants instead of botanists and even then i guess it could be easily countered by wearing a modsuit or whatever else

**Comment by Yobrocharlie** [comment]
soulv

**Comment by moocowswag** [comment]
Won't this make the seed vault pod people's gat fruit WORSE at dealing with lavaland fauna because they dont process chems?

**Comment by mc-oofert** [comment]
> Won't this make the seed vault pod people's gat fruit WORSE at dealing with lavaland fauna because they dont process chems?
> 
> Maybe deal extra damage to mobs and romerol zombies because reference.

not that youre supposed to exit the vault

**Comment by the-orange-cow** [comment]
> not that youre supposed to exit the vault

I don't think pod people have any restriction on leaving the vault.

**Comment by mc-oofert** [comment]
> > not that youre supposed to exit the vault
> 
> I don't think pod people have any restriction on leaving the vault.

not that you can

**Comment by carlarctg** [comment]
> Maybe deal extra damage to mobs and romerol zombies because reference.
pls add zombi bane (check cap sabre for info on how to do that)

**Comment by Loafin34** [comment]
did you modify the gatfruit crop sprite to reflect this new green look? also why would you not make it inject if it has crossbred reagents? it's really funny to have a peashooter with sulfuric acid or whatever those botanists put in it.

**Pull request merged**